src/app/common/ResourceStatus.vue

```diff
@@ -30,15 +38,41 @@ const props = defineProps({
   },
 
   online: {
-    type: Number,
+    type: Number as PropType<number | null>,
     default: null,
   },
 })
+
+const statusAppearance = computed(() => {
+  if (props.online !== null) {
+    const ratio = props.online / props.total
+
+    if (ratio <= 0.5) {
+      return 'danger'
+    } else if (ratio <= 0.85) {
+      return 'warning'
+    }
```

Comment:
How was this picked? To me I feel like we should show warning as soon as `props.online !== props.total`

Reply:
Arbitrarily (and I don’t remember). I changed that to check for `ratio < 1` now.